src/test/scala/PriceListSpec.scala

```diff
@@ -0,0 +1,105 @@
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+
+import com.github.tomakehurst.wiremock.client.WireMock
+import com.github.tomakehurst.wiremock.client.WireMock._
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig
+import com.github.tomakehurst.wiremock.WireMockServer
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.EitherValues
+
+sealed trait PricingError            extends Exception
+case class PriceNotFound(item: Item) extends PricingError
+
+trait PriceList {
+  def priceFor(item: Item): Either[PricingError, PricedItem]
+}
+object PriceList {
+  def apply(prices: Map[Item, Price]): PriceList =
+    (item: Item) => prices.get(item).map(price => PricedItem(item, price)).toRight(PriceNotFound(item))
+}
+
+class PricesFromEqualExperts(baseUrl: String) extends PriceList {
+  override def priceFor(item: Item): Either[PricingError, PricedItem] = {
+    val safeName = URLEncoder.encode(item.name, StandardCharsets.UTF_8.toString)
+    val response = requests.get(s"$baseUrl/backend-take-home-test-data/$safeName.json", check = false)
+    if (response.statusCode == 200) {
+      Right(PricedItem(item, Price(ujson.read(response)("price").num)))
+    } else {
+      Left(PriceNotFound(item))
+    }
+  }
+}
```

Comment:
in my head with this it was just keeping it simple, I guess you might want the signature wrapped in a Future to make it easier to change later if it needed to ever be more than this